src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -767,17 +767,16 @@ public virtual int Read(Span<byte> destination)
 
         // Reads one byte from the stream by calling Read(byte[], int, int). 
         // Will return an unsigned byte cast to an int or -1 on end of stream.
-        // This implementation does not perform well because it allocates a new
-        // byte[] each time you call it, and should be overridden by any 
+        // This implementation should be overridden by any 
         // subclass that maintains an internal buffer.  Then, it can help perf
         // significantly for people who are reading one byte at a time.
         public virtual int ReadByte()
         {
-            byte[] oneByteArray = new byte[1];
-            int r = Read(oneByteArray, 0, 1);
+            Span<byte> bytes = stackalloc byte[1];
+            int r = Read(bytes);
```

Comment:
I'm not sure how useful this is. If someone is overriding the new Read(Span) method, then hopefully they're also overriding ReadByte (and more efficiently). If they're not overriding Read(Span), then the base Read (Span) will fall back to using ArrayPool and a copy, and without measuring it's not clear whether that's actually better than a one-byte allocation.

Reply:
You are right, looking at the asm its pretty terrible. It would be better to rent from the array pool directly rather than stackalloc->span->arraypool. However the granularity is way too high; so it would be better if the caller rented from the array pool and used array overload...

Will revert the Stream change as its dubious

Reply:
Reverted the Stream ReadByte/WriteByte